gnome-initial-setup/gnome-initial-setup-copy-worker.c

```diff
@@ -69,18 +71,47 @@ move_file_from_homedir (GFile       *src_base,
 static void
 unlock_keyring (const gchar *initial_setup_homedir)
 {
-  gchar *file;
-  gchar *password = NULL;
+  g_autofree gchar *file = g_build_filename (initial_setup_homedir, ".config/password", NULL);
+  gsize password_size;
+  g_autofree gchar *password = NULL;
+  SecretValue *secret = NULL;
+  SecretService *service = NULL;
+  g_autoptr(GError) error = NULL;
+
+  if (!g_file_get_contents (file, &password, &password_size, &error)) {
+    g_warning ("Unable to read user password file: %s", error->message);
```

Comment:
(Existing bug, but:) Should we also check whether `password_size > 0`, and error otherwise?

Reply:
No, it is legitimate for a user to have an empty passphrase.

But… haaang on! I can't see where this file is *created*.

Reply:
OK – the file is not created, so all this code was unused, and yet the keyring is still unlocked. I have explained why in a new commit message which deletes this code instead.